content/community/faq.md

```diff
@@ -0,0 +1,149 @@
+```gherkin
+Then the user will be notified
+```
+This way, if the implementation of how the user is notified changes, you will only have to change the step definition (or the helper method called by that step definition),
+rather than the step itself **and** all underlying code.
+
+For more information, have a look at [Writing better Gherkin](/bdd/better-gherkin/) and/or [Writing maintainable automated acceptance tests (pdf).](http://dhemery.com/pdf/writing_maintainable_automated_acceptance_tests.pdf)
+
+## How can I specify my test cases in Excel / csv?
+We advise you *not* to use Excel or csv files to define your test cases; using Excel or csv files is considered an anti-pattern.
+
+One of the goals of Cucumber is to have *executable specifications*. This means your feature files should contain just the right level of information to document the expected behaviour of the system.
+If your test cases are kept in separate files, how would you be able to read the documentation?
+
+This also means you shouldn't have too many details in your feature file. If you do, you might consider moving them to your step definitions or [helper methods](#how-to-use-helper-methods).
+For instance, if you have a form where you need to populate lots of different fields, you might use the [Builder pattern](https://en.wikipedia.org/wiki/Builder_pattern) to do so.
+
+## How can I make Cucumber conditionally skip steps
+Each scenario should test *one thing* and fail for one particular reason. This means there should be no reason to skip steps.
+
+If there does seem to be a reason you'd want to skip steps conditionally, you probably have an anti-pattern.
+For instance, you might be trying to test multiple things in one scenario or you might not be in control of the state of your test environment or test data.
+
+The best thing to do here is to fix the root cause.
+
+## Taking a screenshot after (failed) steps
+Taking a screenshot when a scenario fails, might help you to figure out what went wrong.
+To take a screenshot on failure, you can configure an [after hook](/cucumber/api/#after).
+
+Note that taking a screenshot after *every step* is considered an anti-pattern.
+You should be able to rely on your test automation, without having to check every step of your scenario with a screenshot.
+Your automation should be stable and tests should fail for a clear reason.
+    
+# How do I share state between steps?
+{{% block "ruby,javascript" %}}
+In order to share state between steps, you can use a [world object](/cucumber/state/#world-object).
+{{% /block %}}
+
+{{% block "java,kotlin" %}}
+If you are using Cucumber on the JVM, you can use [dependency injection (DI)](/cucumber/state/#dependency-injection) to share state between steps.
+
+If your project already uses a dependency framework supported by Cucumber (and/or you are familiar with one of them), it's probably easiest to use that framework. 
+Otherwise, Picocontainer is the most light weight framework you can use.
```

Comment:
Maybe should be replaced by `lightest weight`?

Reply:
Hmm, not sure. Using "light weight" as an expression here.

Reply:
Fixed the typos though, thanks for pointing them out!